funcx_sdk/funcx/sdk/client.py

```diff
@@ -449,7 +447,7 @@ def batch_run(self, batch) -> t.List[str]:
 
                 # Checking for 'Failed' is how FuncxResponseError.unpack
                 # originally checked for errors.
-                raise GlobusAPIError(result)
+                raise FuncxTaskExecutionFailed(result.get("reason"))
```

Comment:
We clearly missed this path previously; ought we not include a test?  I'm currently thinking in terms of mocking various responses at the HTTP layer.

Reply:
I'll see if I can add a test without having to mock too much.

Reply:
Test added